pkg/registry/apis/provisioning/resources/folderid.go

```diff
@@ -0,0 +1,76 @@
+package resources
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"path"
+	"regexp"
+	"strings"
+)
+
+var disallowedKubernetesCharacters = regexp.MustCompile(`[^-a-z0-9]`)
+
+// FolderID contains the identifier data for a folder.
+type FolderID struct {
+	// Title is the human-readable name created by a human who wrote it.
+	Title string
+	// KubernetesName represents the name the folder should have, derived from the title.
+	// It contains a suffix calculated from the path of the folder.
+	KubernetesName string
+	// Path is the full path to the folder, as given to the parse function.
+	Path string
+}
+
+func ParseFolderID(dirPath string) FolderID {
```

Comment:
should the same path in two repositories resolve to the same k8s name?

Reply:
Good callout... I'll figure a solution out